activerecord/lib/active_record/persistence.rb

```diff
@@ -561,6 +565,28 @@ def delete(id_or_array)
         delete_by(primary_key => id_or_array)
       end
 
+      # Specifies that the passed-in attributes should be refreshed on create via
+      # appending an SQL +RETURNING+ clause to the +INSERT+ statement.  This can be used
+      # to avoid calling +reload+ and sending an extra +SELECT+ statement under certain
+      # circumstances. (e.g. handling race conditions or database triggers)
+      #
+      # The +attributes+ parameter can be either a single symbol or an array of symbols.
+      #
+      # Note: Columns that are auto-incrementing, virtual, or have a default function do
+      # not need to be specified with this method. They will use +RETURNING+ by default.
+      #
+      # Note: Using this with a database backend that doesn't support +RETURNING+ will
+      # result in an error from malformed queries.
+      #
+      # ==== Examples
+      #
+      #   returning_on_create :subtotal
+      #
+      #   returning_on_create :subtotal, :grand_total
+      def returning_on_create(*attributes)
```

Comment:
Moving away from "returning" sounds good to me. I'll ponder some new naming. "refresh" maybe?  "reload" I strongly associate with `SELECT` statements after years of calling the `reload` method. 

I was thinking we should have two separate configuration attributes between `create` and `update`.  An application could have a trigger on insert but not on update.  It would be simpler though to just do one list.

The sooner we can let the application developer know there's a compatibility problem the better. Should we add a TODO note here to add `validate_returning_support!` at a later time?

Reply:
After reflecting a bit more it is probably better to start simple and do a single macro. We can always add more options later.

I also warmed up to having "reload" in the name. Introducing new vocabulary is going to add more mental overhead than it helps in this case. Most Rails developers are going to immediately understand the implications of "reload" as far as their models are concerned. The goal of ActiveRecord is to abstract away certain database details. They shouldn't have to worry about `RETURNING` vs `SELECT` unless there's a reason to go digging.

So I'm leaning towards the name... `reload_attributes_on_save`

Reply:
>I was thinking we should have two separate configuration attributes between create and update. An application could have a trigger on insert but not on update.

Sorry, I missed your comment and posted similar opinion as a standalone comment. I agree on this, long-term we do need an option to separate `update` and `create`. Ultimately it's up to the core team to make final decision on APIs but generally I believe we aim to design APIs as flexible as possible so we can extend them later without unnecessary deprecations

Reply:
@nvasilevski No worries! I'll probably check in a code change tomorrow.

I'm planning to use [after_commit](https://apidock.com/rails/ActiveRecord/Transactions/ClassMethods/after_commit) as an example for the parameters.  So it will be `reload_attributes :foo, on: [:create, :update]`.  (No destroy obviously)